test/parallel/test-internal-module-map-asserts.js

```diff
@@ -0,0 +1,96 @@
+// Flags: --expose-internals
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const ModuleMap = require('internal/loader/ModuleMap');
+
+// ModuleMap.get, url argument should only accept string values
+{
+  const errorReg = common.expectsError({
+    code: 'ERR_INVALID_ARG_TYPE',
+    type: TypeError,
+    message: /^The "url" argument must be of type string/
+  }, 5);
+
+  const moduleMap = new ModuleMap();
+
+  assert.throws(() => moduleMap.get({}), errorReg);
```

Comment:
Nit: we can put these in a loop to adhere to the DRY principle:

```js
[{}, [], true, 1, () => {}].forEach((element) => {
  assert.throws(() => moduleMap.get(element), errorReg);
});
```

Reply:
These should likely be made into...

```js
common.expectsError(
  () => moduleMap.get([]),
  {
    code: 'ERR_INVALID_ARG_TYPE',
    type: TypeError,
    message: /^The "url" argument must be of type string/
  }
);
```

etc